cache/class-vary-cache.php

```diff
@@ -86,6 +90,11 @@ public static function set_group_for_user( $group, $value ) {
 		// TODO: only send header if we added or changed things
 		// TODO: don't set the cookie if was already set on the request
 		// validate, process $group, etc.
+		if ( strpos( $group, self::GROUP_SEPARATOR ) !== false || strpos( $group, self::VALUE_SEPARATOR ) !== false ||
+			strpos( $value, self::GROUP_SEPARATOR ) !== false || strpos( $value, self::VALUE_SEPARATOR ) !== false ) {
+				trigger_error( 'Cannot use the group or value separator text in the group/value name', E_USER_WARNING );
+			return;
```

Comment:
Should the function return a boolean or a `WP_Error` object now that is has a success/failure result?

Reply:
Good call, I made the change to register_groups as well